EMF2014/Deserializer.cpp

```diff
@@ -32,21 +26,24 @@
  SOFTWARE.
  */
 
-#ifndef _OUTGOING_MESSAGES_MANAGER_H_
-#define _OUTGOING_MESSAGES_MANAGER_H_
-
-#include <Arduino.h>
-#include <FreeRTOS_ARM.h>
-#include "EMF2014Config.h"
+#include "Deserializer.h"
+#include "Serializable.h"
+#include "Utils.h"
+#include "Tilda.h"
 
-class OutgoingMessagesManager {
-public:
-    OutgoingMessagesManager();
+#include "YoApp.h"
 
-    /* This is a blocking method */
-    void handleTransmissionWindow(TickType_t duration);
-private:
+Serializable* Deserializer::deserialize(const byte* data) {
+    uint16_t messageTypeId = data[1] * 256 + data[0];
 
-};
+    if (messageTypeId == MESSAGE_TYPE_ID_YO) {
```

Comment:
Are we going to have an ever growing list of message types to deserialize here? Doesn't seem the right way to go about things. Breaks the open/closed principle: http://en.wikipedia.org/wiki/Open/closed_principle